pkg/controllers/provisioning/nodepool_test.go

```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/aws/karpenter-core/pkg/cloudprovider/fake"
```

Comment:
This PR is a core change that is affecting core scheduling behavior so we absolutely need to make sure that we have a ton of tests around it. Off the top of my head, here's the tests that I think we should have (expanding off the list that you currently have already here):
1. Basic test that ensures that you sum sidecarContainers and regular containers
2. Multiple pods, some pods have native sidecar containers
3. Single pod, single native sidecar containers and a single init container at the end (which has a value that would be larger than the container total)
4. Same test as #3 but now that single init container doesn't exceed the containers sum
5. Single pod, multiple native sidecar containers and a single init container at the end (which has a value that would be larger than the container total)
6. Same test as #5 but now that single init container doesn't exceed the containers sum
7. Single pod, native sidecar containers and init containers interspersed (maybe make like 10 of them) with one that's early in the list exceeding the `sidecarContainers + containers` sum
8. Same as #7 but it doesn't exceed the sum

There's probably other edge cases you can think of but we should make sure that we have a full set of permutations of all of them (you can refer to upstream as well if you need some ideas on the edge cases to validate)